chain/client/src/stateless_validation/processing_tracker.rs

```diff
@@ -0,0 +1,112 @@
+use std::sync::Arc;
+
+use once_cell::sync::OnceCell;
+
+/// `ProcessingDoneTracker` can be used in conjuction with a `ProcessingDoneWaiter`
+/// to wait until some processing is finished. `ProcessingDoneTracker` should be
+/// kept alive as long as the processing is ongoing, then once it's dropped,
+/// the paired `ProcessingDoneWaiter` will be notified that the processing has finished.\
+/// Does NOT implement `Clone`, if you want to clone it, wrap it in an `Arc`.
+pub struct ProcessingDoneTracker(Arc<OnceCell<()>>);
```

Comment:
Some other possible names, `ProcessingFinishedTracker`, `ComputationFinishedTracker`, `ComputationDoneTracker`, `ProcessingEndedTracker`... Opinions?

Reply:
@robin-near @pugachAG @wacban @Longarithm, in case you have opinions.